src/OrchardCore/OrchardCore/Locking/LocalLock.cs

```diff
@@ -35,24 +38,39 @@ public async Task<ILocker> AcquireLockAsync(string key, TimeSpan? expiration = n
     /// </summary>
     public async Task<(ILocker locker, bool locked)> TryAcquireLockAsync(string key, TimeSpan timeout, TimeSpan? expiration = null)
     {
+        ArgumentException.ThrowIfNullOrWhiteSpace(key);
+        ObjectDisposedException.ThrowIf(_disposed, this);
+
         var semaphore = GetOrCreateSemaphore(key);
 
-        if (await semaphore.Value.WaitAsync(timeout != TimeSpan.MaxValue ? timeout : Timeout.InfiniteTimeSpan))
+        var effectiveTimeout = timeout != TimeSpan.MaxValue ? timeout : Timeout.InfiniteTimeSpan;
```

Comment:
According to the method's docs, `null` should be treated as `Timeout.InfiniteTimeSpan` too.

Reply:
That's for the `expiration`, `timeout` cannot be null here (it's a value type).

Reply:
Ah, my bad.